### PR TITLE
Refactors Money

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Refactors money

This should make money a bit easier to use; instead of being based on amounts *and* bills, it's just universally fungible. What do I mean by that? It means that you can combine any amount of money with any other amount of money, up to a cap of 1,000,000. This makes it far easier to combine, give, or distribute ***specific*** amounts of money to other individuals.

Code wise, this makes more sense; instead of it being some snowflaked thing, it uses the pre-existing stack system.

Icons from Goonstation (flashing one is 1,000,000 credits)

img

:cl: Fox McCloud
tweak: money is now far easier to use, and distribute in amounts you would like
/:cl:

### CODE REVIEW

File: _maps/map_files/RandomZLevels/beach.dmm [hunk 1]
```diff
@@ -2,7 +2,7 @@
 "ab" = (/turf/unsimulated/beach/water/deep,/area/awaymission/undersea)
 "ac" = (/obj/machinery/poolcontroller/seacontroller,/turf/unsimulated/beach/water/deep,/area/awaymission/undersea)
 "ad" = (/turf/unsimulated/beach/water/deep/rock_wall,/area/awaymission/undersea)
-"ae" = (/turf/unsimulated/beach/water/deep/sand_floor,/area/awaymission/undersea)
+"ae" = (/obj/item/stack/spacecash/c500,/turf/unsimulated/beach/water/deep/sand_floor,/area/awaymission/undersea)
```
Comment (Fox-McCloud) [inline _maps/map_files/RandomZLevels/beach.dmm]: This was map merged, believe it or not.

File: code/modules/arcade/arcade_base.dm [hunk 2]
```diff
@@ -83,16 +83,12 @@
 /obj/machinery/arcade/update_icon()
 	return
 
-/obj/machinery/arcade/proc/pay_with_cash(var/obj/item/weapon/spacecash/cashmoney, var/mob/user)
-	if(cashmoney.get_total() < token_price)
+/obj/machinery/arcade/proc/pay_with_cash(obj/item/stack/spacecash/cashmoney, mob/user)
```
Comment (Fox-McCloud) [inline code/modules/arcade/arcade_base.dm]: Damn you and your copy-pasta @FalseIncarnate

**Comment by TravellingMerchant** [comment]
Damn,  an awesome change I never knew I needed.  Can't wait for this!

**Comment by SamHPurp** [comment]
You can now split stacks of cash? If so... ❤️

**Comment by KasparoVy** [comment]
DOSH, grab it while you can lads

**Comment by Birdtalon** [comment]
Next we just need an actual use for money.

**Comment by Fox-McCloud** [comment]
https://soundcloud.com/audiodile/abc-how-to-be-a-millionaire-audiodiles-bankrupt-mix

**Pull request merged**